File: Fecha.cpp
// Archivo Fecha.cpp

#include "Fecha.h"

Fecha::Fecha() : Dia(9), Mes(10), Agno(2023) {}

Fecha::Fecha(int dia, int mes, int agno) : Dia(dia), Mes(mes), Agno(agno) {}

int Fecha::getDia(){
	return Dia;
}

int Fecha::getMes(){
	return Mes;
}

int Fecha::getAgno(){
	return Agno;
}

void Fecha::setDia(int dia) {
	Dia = dia;
}

void Fecha::setMes(int mes) {
	Mes = mes;
}

void Fecha::setAgno(int agno) {
	Agno = agno;
}

Fecha Fecha::Get_Fecha(){
	return *this;
}

bool Fecha::operator==(Fecha* fecha){
    return (Dia == fecha->getDia()) && (Mes == fecha->getMes()) && (Agno == fecha->getAgno());
}


void Fecha::Show_Fecha(){
	cout << Dia << "/" << Mes << "/" << Agno << endl;
}

void Fecha::AddDia(){
	Dia ++;
	if(Dia >= 31){
		Dia = 1;
		Mes++;
	}
	if(Mes >= 12 && Dia >= 31){
		Mes = 1;
		Dia = 1;
		Agno++;
	}
}

File: fecha.py
class Fecha:
    def __init__(self, dia=9, mes=10, agno=2023):
        self.Dia = dia
        self.Mes = mes
        self.Agno = agno

    def getDia(self):
        return self.Dia

    def getMes(self):
        return self.Mes

    def getAgno(self):
        return self.Agno

    def setDia(self, dia):
        self.Dia = dia

    def setMes(self, mes):
        self.Mes = mes

    def setAgno(self, agno):
        self.Agno = agno

    def Get_Fecha(self):
        return self

    def __eq__(self, fecha):
        return (self.Dia == fecha.getDia()) and (self.Mes == fecha.getMes()) and (self.Agno == fecha.getAgno())

    def Show_Fecha(self):
        print(f"{self.Dia}/{self.Mes}/{self.Agno}")

    def AddDia(self):
        self.Dia += 1
        if self.Dia >= 31:
            self.Dia = 1
            self.Mes += 1
        if self.Mes >= 12 and self.Dia >= 31:
            self.Mes = 1
            self.Dia = 1
            self.Agno += 1
